File: CKS.Prototype/Scripts/Entities/project.js
var cks = cks || {};
cks.entities = cks.entities || {};
cks.entities.project = function (planningVm, traceEstimateVm, wallsVm, sectionsEntity) {
    var _toJs = function () {
        return {
            title: planningVm.title(),
            systemOfMeasurement: planningVm.unitsSystem(),
            sections: sectionsEntity.toJs(),
            createdDate: planningVm.createdDate(),
            id: planningVm.id()
        };
    };

    var _toVm = function (project) {
        if (!project) {
            planningVm.title(undefined);
            planningVm.unitsSystem(undefined);
            sectionsEntity.toVm(undefined);
            planningVm.createdDate(undefined);
            planningVm.id(undefined);
            return;
        }
        planningVm.title(project.title);
        planningVm.unitsSystem(project.systemOfMeasurement);
        sectionsEntity.toVm(project.sections);
        planningVm.createdDate(project.createdDate);
        planningVm.id(project.id);
    };
    return {
        toJs: _toJs,
        toVm: _toVm
    };
};

File: CKS.Entities/Project.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace CKS.Entities
{
    public class Project: BaseEntity
    {
        public string Title { get; set; }
        public string SystemOfMeasurement { get; set; }
        [JsonIgnore]
        public virtual UserProfile UserProfile { get; set; }
        public virtual ICollection<Section> Sections { get; set; }
        public virtual ICollection<TypeOfWork> TypesOfWork { get; set; }
    }
}
